PublicMapping/districtbuilder
# Update server and management node dependencies

## Overview

Updates Node, Nest.JS (and related plugins like `nestjs/crud`), TypeORM, `eslint`, and `prettier` for the `server` and `manage` modules.

The main intent here was to update our version of TypeORM, which was generating spurious migrations related to the `pinned_metric_fields` column whenever generating new migrations (requiring editing the new migrations to remove those lines), but doing so pulled on a long chain of dependencies that needed to be updated - the main one being that updating Nest.JS moved us from TypeScript 3 -> 4 and Node 12 -> 16 (I think we could have jumped to only 14, but I choose to go to the most recent LTS), which then required updating a number of developer tools like `eslint` and `prettier`.

### Checklist



## Testing Instructions

- `scripts/update`
- `scripts/server`
  - Test out the application and ensure there haven't been any regressions
- Spot check the `scripts/manage` commands for regressions
  - Process & publish geojson for a small region
  - `./scripts/manage create-random-projects 1 all`
  - etc.

Closes #891 
Closes #210
